src/admin/dashboard/pages/dashboard/sections/MostReviewedProductsSection/index.tsx

```diff
@@ -0,0 +1,124 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import Section from '../../Elements/Section';
+import { DataViews } from '../../../../../../components';
+import { useDashboardApiData } from '../../hooks/useDashboardApiData';
+import { fetchMostReviewedProducts } from '../../utils/api';
+import { MostReviewedProductsData } from '../../types';
+import MostReviewedProductsSkeleton from './Skeleton';
+import { applyFilters } from '@wordpress/hooks';
+
+const MostReviewedProductsSection = () => {
+    const { data, loading, error } =
+        useDashboardApiData< MostReviewedProductsData >( {
+            fetchFunction: fetchMostReviewedProducts,
+        } );
+
+    const padDefaultData = ( originalData ) => {
+        const paddedData = [ ...originalData.filter( data => data.review_count ) ];
+
+        // If the data is empty, fill with default values.
+        const emptyString = applyFilters(
+            'dokan_admin_dashboard_most_reviewed_default_table_data',
+            __( '--', 'dokan-lite' ),
+            originalData
+        );
+
+        // Add empty rows with -- if we have less than 5 items.
+        while ( paddedData.length < 5 ) {
+            paddedData.push( {
+                rank: emptyString,
+                product_title: emptyString,
+                review_count: emptyString,
+            } );
+        }
```

Comment:
I think that we should limit the range in frontend where we are limiting in backend. We should display all data those are fetched by the API.